extensions/ql-vscode/src/queries-panel/query-tree-view-item.ts

```diff
@@ -1,11 +1,17 @@
 import * as vscode from "vscode";
 
-export class QueryTreeViewItem extends vscode.TreeItem {
+export abstract class QueryTreeViewItem extends vscode.TreeItem {
```

Comment:
Why do we need the abstract class?

Reply:
The reason for making this class abstract, even though it doesn't have any abstract methods, is that there shouldn't be any need to construct an instance of it directly. You should always be using one of the subclasses.

By making it abstract we guide future developers to make the right choice by using one of the subclasses and not accidentally use this class for something it wasn't designed for.

Reply:
Of course, that makes sense, thanks!